app/controllers/sign_up/recovery_codes_controller.rb

```diff
@@ -35,6 +35,8 @@ def next_step
         sign_up_completed_path
       elsif current_user.password_reset_profile.present?
         reactivate_profile_path
+      elsif (stored_location = stored_location_for(current_user))
```

Comment:
Yeah I found that, but this method seemed to have some special logic already (ex the password reset profile part) so I think this is the right choice